### PR TITLE
use node-gyp-pre for prebuilt binaries

### PR DESCRIPTION
**Pull request by blackmad**
- all tests pass in our circleci

- tested in a simple project, observed no compiling + status messages about download binary
image

- compiled a version of large real project on docker/linux/circle with s2 pointing to this branch,
---> npm install --save radarlabs/s2#getting-ready-for-node-pre-gyp
 observed
[@radarlabs/s2] Success: "/app/node_modules/@radarlabs/s2/lib/binding/Release/node-v72-linux-x64/s2.node" is installed via remote
observed all tests passing

### CODE REVIEW

**Comment by blackmad** [comment]
lol should have squashed before this review

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,60 +1,43 @@
 version: 2
 
+dockerbuild: &dockerbuild
```
Comment (jkao) [inline .circleci/config.yml]: what does the `&dockerbuild` token mean

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,60 +1,43 @@
...
+  steps:
+    - checkout
+    - run:
+        name: "Pull submodules"
+        command: |
```
Comment (jkao) [inline .circleci/config.yml]: also this pipe 🤔

File: index.js [hunk 3]
```diff
@@ -1,2 +1,7 @@
-const s2 = require('bindings')('s2');
+// magic incantation from step 3 @ https://github.com/mapbox/node-pre-gyp#readme
+const binary = require('node-pre-gyp');
+const path = require('path');
+var binding_path = binary.find(path.resolve(path.join(__dirname,'./package.json')));
```
Comment (jkao) [inline index.js]: trying to understand why we need this now?

File: test.sh [hunk 4]
```diff
@@ -20,3 +20,8 @@ docker build -t $NODE_11_IMAGE -f $NODE_11_DOCKERFILE . && docker run $NODE_11_I
 NODE_12_IMAGE=test12
 NODE_12_DOCKERFILE=./docker/Dockerfile.node12.test
 docker build -t $NODE_12_IMAGE -f $NODE_12_DOCKERFILE . && docker run $NODE_12_IMAGE:latest
+
+# test node 13
+NODE_13_IMAGE=test13
+NODE_13_DOCKERFILE=./docker/Dockerfile.node13.test
+docker build -t $NODE_13_IMAGE -f $NODE_13_DOCKERFILE . && docker run $NODE_12_IMAGE:latest
```
Comment (jkao) [inline test.sh]: I think there's a typo 12 -> 13

File: test/Cell.test.js [hunk 5]
```diff
@@ -1,4 +1,8 @@
-const s2 = require('bindings')('s2');
+// magic incantation from step 3 @ https://github.com/mapbox/node-pre-gyp#readme
+const binary = require('node-pre-gyp');
+const path = require('path');
+var binding_path = binary.find(path.resolve('./package.json'));
+const s2 = require(binding_path);
```
Comment (jkao) [inline test/Cell.test.js]: hmm gotta say I'm not a fan of this expansion for our tests - does the old way not work anymore? what are the path differences?

File: .circleci/config.yml (hunk 1 above)
Comment (blackmad) [reply .circleci/config.yml]: it makes it an anchor to be referenced later

File: .circleci/config.yml (hunk 2 above)
Comment (blackmad) [reply .circleci/config.yml]: multiline

File: index.js (hunk 3 above)
Comment (blackmad) [reply index.js]: magic incantation. see the comment.

File: test.sh (hunk 4 above)
Comment (blackmad) [reply test.sh]: thanks, fixed

File: test/Cell.test.js (hunk 5 above)
Comment (blackmad) [reply test/Cell.test.js]: I will try to make it simpler, stand by

File: index.js (hunk 3 above)
Comment (blackmad) [reply index.js]: okay, here's an actual answer to both this and the next Q (about tests)

we are dynamically requiring the bindings for our particular platform. This incantation finds that via node-gyp-pre's logic, see below:

Welcome to Node.js v13.6.0.
Type ".help" for more information.
> const binary = require('node-pre-gyp');
undefined
>
> var binding_path = binary.find(path.resolve('./package.json'));
undefined
>
> binding_path
'/Users/blackmad/RadarCode/s2/lib/binding/Release/node-v79-darwin-x64/s2.node'
>

File: test/Cell.test.js (hunk 5 above)
Comment (blackmad) [reply test/Cell.test.js]: see above for reason. I can't really make this any simpler.

File: manual-publish.sh [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+set -e
+
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+
+nvm install v10
+nvm use v10
+rm -rf node_modules
+npm install
+JOBS=max npx node-pre-gyp rebuild package unpublish publish
```
Comment (jkao) [inline manual-publish.sh]: this look at `~/.aws/credentials` right?

**Comment by jkao** [comment]
just one more question on how the manual-publish script works. looks okay to me. we'll have to cut a 0.0.3 build on npm

**Comment by blackmad** [comment]
node-pre-gyp publish looks at env vars  node_pre_gyp_accessKeyId=
node_pre_gyp_secretAccessKey=
- want me to add a comment?

**Pull request merged**